realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -230,6 +232,25 @@ private static synchronized void removeSession(SyncConfiguration syncConfigurati
         }
     }
 
+    /**
+     * Retruns the all valid sessions belonging to the user.
+     *
+     * @param syncUser the user to use.
+     * @return the all valid sessions belonging to the user.
+     */
+    static synchronized List<SyncSession> getAllSessions(SyncUser syncUser) {
```

Comment:
I think you get rid of the synchronization on this method.
`sessions` is already concurrent.
... and synchronizing this method does *nothing* if `getState` or `getUser` returns mutable state...